### PR TITLE
fix(types): fixes to @socketsupply/socket types

### PR DESCRIPTION
**Pull request by aleclarson**
Found some broken types and made `Extension` generic so I could specify its `binding` type

### CODE REVIEW

**Review by jwerle — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
@aleclarson please run `npm run gen` to regenerate documentation as well

File: bin/docs-generator/api-module.js [hunk 1]
```diff
@@ -141,21 +141,21 @@ export function generateApiModuleDoc ({
...
-          const [name, defaultValue] = rawName.replace(/[[\]']+/g, '').trim().split('=')
+          const [name, defaultValue] = rawName.replace(/[[\]']+/g, '').trim().split(/ *[=] */)
 
           // type could be [(string|number)=]
           const parenthasisedType = rawType
             .replace(/\s*\|\s*/g, ' \\| ')
             .replace(/\[|\]/g, '')
           // now it is (string|number)=
-          const optional = parenthasisedType.endsWith('=')
+          const optional = parenthasisedType.endsWith('=') || /^ *\[/.test(rawName)
```
Comment (aleclarson) [inline bin/docs-generator/api-module.js:151]: added handling for `[optionalParam]` without default value

File: bin/docs-generator/api-module.js [hunk 2]
```diff
@@ -141,21 +141,21 @@ export function generateApiModuleDoc ({
           item.signature = item.signature || []
           const parts = attr.split(/-\s+(.*)/)
           const { 1: rawType, 2: rawName } = parts[0].match(/{([^}]+)}(.*)/)
-          const [name, defaultValue] = rawName.replace(/[[\]']+/g, '').trim().split('=')
+          const [name, defaultValue] = rawName.replace(/[[\]']+/g, '').trim().split(/ *[=] */)
```
Comment (aleclarson) [inline bin/docs-generator/api-module.js:144]: added whitespace tolerance to param name `=` splitting

**Review by chicoxyzzy — APPROVED** [review]
👍

**Pull request merged**